radicale/__init__.py

```diff
@@ -425,6 +433,13 @@ def response(status, headers=(), answer=None):
                     environ, base_prefix, path, user)
             except socket.timeout:
                 return response(*REQUEST_TIMEOUT)
+            except vobject.base.VObjectError as e:
+                return response(*UNPROCESSABLE_ENTITY(e))
+            except xmlutils.ET.ParseError as e:
+                return response(*UNPROCESSABLE_ENTITY("Invalid XML: %s" % e))
+            except:
```

Comment:
Hmm, I don't think a thread will ever get a KeyboardInterrupt, but I can handle this case, if you like.